ste/downloader-blob.go

```diff
@@ -59,6 +68,19 @@ func (bd *blobDownloader) Epilogue() {
 func (bd *blobDownloader) GenerateDownloadFunc(jptm IJobPartTransferMgr, srcPipeline pipeline.Pipeline, destWriter common.ChunkedFileWriter, id common.ChunkID, length int64, pacer pacer) chunkFunc {
 	return createDownloadChunkFunc(jptm, id, func() {
 
+		// If the range does not contain any data, write out empty data to disk without performing download
+		if bd.pageRangeOptimizer != nil && !bd.pageRangeOptimizer.doesRangeContainData(
+			azblob.PageRange{Start: id.OffsetInFile(), End: id.OffsetInFile() + length - 1}) {
+
+			// TODO confirm this is the right wait reason
```

Comment:
Yes, Body is fine. Or you can probably just not set it. Because chunked file writer will immediately change it anyway.

Reply:
ok